// element-plus/lib/utils/vue/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var globalNode = require('./global-node.js');
var icon = require('./icon.js');
var install = require('./install.js');
var props = require('./props.js');
var refs = require('./refs.js');
var style = require('./style.js');
require('./typescript.js');
var validator = require('./validator.js');
var vnode = require('./vnode.js');



exports.changeGlobalNodesTarget = globalNode.changeGlobalNodesTarget;
exports.createGlobalNode = globalNode.createGlobalNode;
exports.removeGlobalNode = globalNode.removeGlobalNode;
exports.CloseComponents = icon.CloseComponents;
exports.TypeComponents = icon.TypeComponents;
exports.TypeComponentsMap = icon.TypeComponentsMap;
exports.ValidateComponentsMap = icon.ValidateComponentsMap;
exports.iconPropType = icon.iconPropType;
exports.withInstall = install.withInstall;
exports.withInstallFunction = install.withInstallFunction;
exports.withNoopInstall = install.withNoopInstall;
exports.buildProp = props.buildProp;
exports.buildProps = props.buildProps;
exports.definePropType = props.definePropType;
exports.propKey = props.propKey;
exports.composeRefs = refs.composeRefs;
exports.addUnit = style.addUnit;
exports.isValidComponentSize = validator.isValidComponentSize;
exports.isValidDatePickType = validator.isValidDatePickType;
exports.PatchFlags = vnode.PatchFlags;
exports.ensureOnlyChild = vnode.ensureOnlyChild;
exports.getFirstValidNode = vnode.getFirstValidNode;
exports.getNormalizedProps = vnode.getNormalizedProps;
exports.isComment = vnode.isComment;
exports.isFragment = vnode.isFragment;
exports.isTemplate = vnode.isTemplate;
exports.isText = vnode.isText;
exports.isValidElementNode = vnode.isValidElementNode;
exports.renderBlock = vnode.renderBlock;
exports.renderIf = vnode.renderIf;
