// element-plus/lib/utils/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('./dom/index.js');
require('./vue/index.js');
var animation = require('./animation.js');
var arrays = require('./arrays.js');
var browser = require('./browser.js');
var error = require('./error.js');
var i18n = require('./i18n.js');
var objects = require('./objects.js');
var raf = require('./raf.js');
var rand = require('./rand.js');
var strings = require('./strings.js');
var types = require('./types.js');
var typescript = require('./typescript.js');
var aria = require('./dom/aria.js');
var event = require('./dom/event.js');
var position = require('./dom/position.js');
var resizeEvent = require('./dom/resize-event.js');
var scroll = require('./dom/scroll.js');
var style = require('./dom/style.js');
var globalNode = require('./vue/global-node.js');
var icon = require('./vue/icon.js');
var install = require('./vue/install.js');
var props = require('./vue/props.js');
var refs = require('./vue/refs.js');
var style$1 = require('./vue/style.js');
var validator = require('./vue/validator.js');
var vnode = require('./vue/vnode.js');
var lodashUnified = require('lodash-unified');
var shared = require('@vue/shared');
var core = require('@vueuse/core');
var vue = require('vue');



exports.cubic = animation.cubic;
exports.easeInOutCubic = animation.easeInOutCubic;
exports.castArray = arrays.castArray;
exports.unique = arrays.unique;
exports.isFirefox = browser.isFirefox;
exports.debugWarn = error.debugWarn;
exports.throwError = error.throwError;
exports.isKorean = i18n.isKorean;
exports.entriesOf = objects.entriesOf;
exports.getProp = objects.getProp;
exports.keysOf = objects.keysOf;
exports.cAF = raf.cAF;
exports.rAF = raf.rAF;
exports.generateId = rand.generateId;
exports.getRandomInt = rand.getRandomInt;
exports.escapeStringRegexp = strings.escapeStringRegexp;
exports.isElement = types.isElement;
exports.isEmpty = types.isEmpty;
exports.isPropAbsent = types.isPropAbsent;
exports.isUndefined = types.isUndefined;
exports.mutable = typescript.mutable;
exports.attemptFocus = aria.attemptFocus;
exports.focusNode = aria.focusNode;
exports.getSibling = aria.getSibling;
exports.isFocusable = aria.isFocusable;
exports.isLeaf = aria.isLeaf;
exports.isVisible = aria.isVisible;
exports.obtainAllFocusableElements = aria.obtainAllFocusableElements;
exports.triggerEvent = aria.triggerEvent;
exports.composeEventHandlers = event.composeEventHandlers;
exports.off = event.off;
exports.on = event.on;
exports.once = event.once;
exports.whenMouse = event.whenMouse;
exports.getClientXY = position.getClientXY;
exports.getOffsetTop = position.getOffsetTop;
exports.getOffsetTopDistance = position.getOffsetTopDistance;
exports.isInContainer = position.isInContainer;
exports.addResizeListener = resizeEvent.addResizeListener;
exports.removeResizeListener = resizeEvent.removeResizeListener;
exports.getScrollBarWidth = scroll.getScrollBarWidth;
exports.getScrollContainer = scroll.getScrollContainer;
exports.isScroll = scroll.isScroll;
exports.scrollIntoView = scroll.scrollIntoView;
exports.addClass = style.addClass;
exports.classNameToArray = style.classNameToArray;
exports.getStyle = style.getStyle;
exports.hasClass = style.hasClass;
exports.removeClass = style.removeClass;
exports.removeStyle = style.removeStyle;
exports.setStyle = style.setStyle;
exports.changeGlobalNodesTarget = globalNode.changeGlobalNodesTarget;
exports.createGlobalNode = globalNode.createGlobalNode;
exports.removeGlobalNode = globalNode.removeGlobalNode;
exports.CloseComponents = icon.CloseComponents;
exports.TypeComponents = icon.TypeComponents;
exports.TypeComponentsMap = icon.TypeComponentsMap;
exports.ValidateComponentsMap = icon.ValidateComponentsMap;
exports.iconPropType = icon.iconPropType;
exports.withInstall = install.withInstall;
exports.withInstallFunction = install.withInstallFunction;
exports.withNoopInstall = install.withNoopInstall;
exports.buildProp = props.buildProp;
exports.buildProps = props.buildProps;
exports.definePropType = props.definePropType;
exports.propKey = props.propKey;
exports.composeRefs = refs.composeRefs;
exports.addUnit = style$1.addUnit;
exports.isValidComponentSize = validator.isValidComponentSize;
exports.isValidDatePickType = validator.isValidDatePickType;
exports.PatchFlags = vnode.PatchFlags;
exports.ensureOnlyChild = vnode.ensureOnlyChild;
exports.getFirstValidNode = vnode.getFirstValidNode;
exports.getNormalizedProps = vnode.getNormalizedProps;
exports.isComment = vnode.isComment;
exports.isFragment = vnode.isFragment;
exports.isTemplate = vnode.isTemplate;
exports.isText = vnode.isText;
exports.isValidElementNode = vnode.isValidElementNode;
exports.renderBlock = vnode.renderBlock;
exports.renderIf = vnode.renderIf;
Object.defineProperty(exports, 'ensureArray', {
	enumerable: true,
	get: function () { return lodashUnified.castArray; }
});
Object.defineProperty(exports, 'camelize', {
	enumerable: true,
	get: function () { return shared.camelize; }
});
Object.defineProperty(exports, 'capitalize', {
	enumerable: true,
	get: function () { return shared.capitalize; }
});
Object.defineProperty(exports, 'hasOwn', {
	enumerable: true,
	get: function () { return shared.hasOwn; }
});
Object.defineProperty(exports, 'hyphenate', {
	enumerable: true,
	get: function () { return shared.hyphenate; }
});
Object.defineProperty(exports, 'isArray', {
	enumerable: true,
	get: function () { return shared.isArray; }
});
Object.defineProperty(exports, 'isDate', {
	enumerable: true,
	get: function () { return shared.isDate; }
});
Object.defineProperty(exports, 'isFunction', {
	enumerable: true,
	get: function () { return shared.isFunction; }
});
Object.defineProperty(exports, 'isObject', {
	enumerable: true,
	get: function () { return shared.isObject; }
});
Object.defineProperty(exports, 'isPromise', {
	enumerable: true,
	get: function () { return shared.isPromise; }
});
Object.defineProperty(exports, 'isString', {
	enumerable: true,
	get: function () { return shared.isString; }
});
Object.defineProperty(exports, 'isSymbol', {
	enumerable: true,
	get: function () { return shared.isSymbol; }
});
Object.defineProperty(exports, 'kebabCase', {
	enumerable: true,
	get: function () { return shared.hyphenate; }
});
Object.defineProperty(exports, 'isBoolean', {
	enumerable: true,
	get: function () { return core.isBoolean; }
});
Object.defineProperty(exports, 'isNumber', {
	enumerable: true,
	get: function () { return core.isNumber; }
});
Object.defineProperty(exports, 'isVNode', {
	enumerable: true,
	get: function () { return vue.isVNode; }
});
