// element-plus/lib/utils/dom/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var aria = require('./aria.js');
var event = require('./event.js');
var position = require('./position.js');
var resizeEvent = require('./resize-event.js');
var scroll = require('./scroll.js');
var style = require('./style.js');



exports.attemptFocus = aria.attemptFocus;
exports.focusNode = aria.focusNode;
exports.getSibling = aria.getSibling;
exports.isFocusable = aria.isFocusable;
exports.isLeaf = aria.isLeaf;
exports.isVisible = aria.isVisible;
exports.obtainAllFocusableElements = aria.obtainAllFocusableElements;
exports.triggerEvent = aria.triggerEvent;
exports.composeEventHandlers = event.composeEventHandlers;
exports.off = event.off;
exports.on = event.on;
exports.once = event.once;
exports.whenMouse = event.whenMouse;
exports.getClientXY = position.getClientXY;
exports.getOffsetTop = position.getOffsetTop;
exports.getOffsetTopDistance = position.getOffsetTopDistance;
exports.isInContainer = position.isInContainer;
exports.addResizeListener = resizeEvent.addResizeListener;
exports.removeResizeListener = resizeEvent.removeResizeListener;
exports.getScrollBarWidth = scroll.getScrollBarWidth;
exports.getScrollContainer = scroll.getScrollContainer;
exports.isScroll = scroll.isScroll;
exports.scrollIntoView = scroll.scrollIntoView;
exports.addClass = style.addClass;
exports.classNameToArray = style.classNameToArray;
exports.getStyle = style.getStyle;
exports.hasClass = style.hasClass;
exports.removeClass = style.removeClass;
exports.removeStyle = style.removeStyle;
exports.setStyle = style.setStyle;
